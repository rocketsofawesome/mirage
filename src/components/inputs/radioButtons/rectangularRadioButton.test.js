import React from 'react'
import 'jest-styled-components'

import { RectangularRadioButton } from 'SRC'

const { shallowWithTheme } = global

const input = {
  defaultChecked: false,
  name: 'gender',
  labels: 'BOYS',
  value: 'boys',
  input: { onChange: () => {} }
}

describe('(Styled Component) RectangularRadio', () => {
  const createRectangularRadio = (props) => {
    let inputProp = input
    if (props && props.input) {
      inputProp = {...input, ...props.input}
    }
    return shallowWithTheme(<RectangularRadioButton {...props} input={inputProp} />)
  }
  test('matching the snapshot', () => {
    expect(createRectangularRadio())
    .toMatchSnapshot()
  })
})
