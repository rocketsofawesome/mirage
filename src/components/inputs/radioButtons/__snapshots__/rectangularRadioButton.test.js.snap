// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Styled Component) RectangularRadio matching the snapshot 1`] = `
.c0 {
  position: relative;
}

.c0 > input {
  position: absolute;
  width: 0;
  height: 0;
  opacity: 0;
}

.c0 > label {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 10vw;
  width: 10vw;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  padding-left: 1rem;
  padding-right: 1rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  cursor: pointer;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  -webkit-transition-duration: 0.25s;
  transition-duration: 0.25s;
  -webkit-transition-property: background-color,border-color,color;
  transition-property: background-color,border-color,color;
  text-align: center;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  font-family: "ff-din-web",Helvetica,Arial,sans-serif;
  font-size: 1.6rem;
  font-weight: 500;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  background-color: #F3F3F3;
  border: 1px solid;
  border-color: #D5D5D5;
  color: #00003C;
  height: 6.4rem;
  width: 100%;
}

.c0 > input:checked + label {
  background-color: #00003C;
  color: #FFFFFF;
}

<BaseRectangularRadioButton
  className="c0"
  error={null}
  name="gender"
  shape="rectangle"
  value="boys"
/>
`;
